functests-performance-profile-creator/1_performance-profile_creator/ppc.go

```diff
@@ -0,0 +1,159 @@
+		mustGatherDirs := getMustGatherDirs(mustGatherPath)
+		// full profile path => arguments the profile was created with
+		expectedProfiles := getExpectedProfiles(expectedProfilesPath, mustGatherDirs)
+
+		for expectedProfilePath, args := range expectedProfiles {
+			cmdArgs := fmt.Sprintf("--disable-ht=%v --mcp-name=%s --must-gather-dir-path=%s "+
+				"--power-consumption-mode=%s --profile-name=%s --reserved-cpu-count=%d --rt-kernel=%v "+
+				"--split-reserved-cpus-across-numa=%v --topology-manager-policy=%s --user-level-networking=%v",
+				args.DisableHT, args.MCPName, args.MustGatherDirPath, args.PowerConsumptionMode,
+				args.ProfileName, args.ReservedCPUCount, args.RTKernel, args.SplitReservedCPUsAcrossNUMA,
+				args.TMPolicy, args.UserLevelNetworking)
+			cmdArgsSplit := strings.Split(cmdArgs, " ")
+
+			out, err := testutils.ExecAndLogCommand(ppcPath, cmdArgsSplit...)
+			Expect(err).To(BeNil(), "failed to run ppc for '%s': %v", expectedProfilePath, err)
+
+			profile := &performancev2.PerformanceProfile{}
+			err = yaml.Unmarshal(out, profile)
+			Expect(err).To(BeNil(), "failed to unmarshal the output yaml for '%s': %v", expectedProfilePath, err)
+
+			bytes, err := ioutil.ReadFile(expectedProfilePath)
+			Expect(err).To(BeNil(), "failed to read the expected yaml for '%s': %v", expectedProfilePath, err)
+
+			expectedProfile := &performancev2.PerformanceProfile{}
+			err = yaml.Unmarshal(bytes, expectedProfile)
+			Expect(err).To(BeNil(), "failed to unmarshal the expected yaml for '%s': %v", expectedProfilePath, err)
+
+			Expect(profile).To(BeEquivalentTo(expectedProfile), "regression test failed for '%s' case", expectedProfilePath)
+		}
+	})
+})
+
+func getMustGatherDirs(mustGatherPath string) map[string]string {
+	Expect(mustGatherPath).To(BeADirectory())
+
+	mustGatherDirs := make(map[string]string)
+	mustGatherPathContent, err := ioutil.ReadDir(mustGatherPath)
+	Expect(err).To(BeNil(), fmt.Errorf("can't list '%s' files: %v", mustGatherPath, err))
+
+	for _, file := range mustGatherPathContent {
+		fullFilePath := filepath.Join(mustGatherPath, file.Name())
+		Expect(fullFilePath).To(BeADirectory())
+
+		mustGatherDirs[file.Name()] = fullFilePath
+	}
+
+	return mustGatherDirs
+}
+
+func getExpectedProfiles(expectedProfilesPath string, mustGatherDirs map[string]string) map[string]cmd.ProfileCreatorArgs {
+	Expect(expectedProfilesPath).To(BeADirectory())
+
+	expectedProfiles := make(map[string]cmd.ProfileCreatorArgs)
+	expectedProfilesPathContent, err := ioutil.ReadDir(expectedProfilesPath)
+	Expect(err).To(BeNil(), fmt.Errorf("can't list '%s' files: %v", expectedProfilesPath, err))
+
+	for _, file := range expectedProfilesPathContent {
+		fullFilePath := filepath.Join(expectedProfilesPath, file.Name())
+		Expect(filepath.Ext(fullFilePath)).To(Equal(".yaml"))
+
+		var ppcArgs cmd.ProfileCreatorArgs
+
+		args := strings.Split(strings.TrimSuffix(file.Name(), filepath.Ext(file.Name())), argsDelimiter)
```

Comment:
what's the benefit of encoding the expected values in the file name?

Reply:
Each profile yaml file will by a self contained test (input/output)
I added 2 use-cases in the PR

Reply:
Sure, but why, for example, we cannot read back these values from the profile themselves? IOW, why should we encode this data *in the file names*? This is pretty much like adding a little DSL in disguise, and I'm trying to understand if we have simpler alternatives. Like just reading the manifests themselves :)

Reply:
This would a be a good idea if we would have been able to "easily" map profile values to ppc arguments, however this is not the case. Tracing back a certain "isolated cpus"/"reserved cpus" partition to --reserved-cpus=? -split-reserved-cpus-across-numa-nodes=? ht-disabled=? is far from trivial and the complexity is "almost" as the original code.   The drawback is that a failed e2e test may be caused by the test implementation and not the tool itself.

While I do agree the args-encoded-in-file-name has a "smell", is not really a DSL, is a simple key-value list and will bever go beyond that. The last point is we want to make it easy for the QE to map their testing "matrix" (args combination -> profile) easier.

If you find this encoding as a  "no-go", I can add a file that defines the mappings, it will basically look the same, a list of key-value pairs pointing to a yaml file.

Reply:
good point. So what about a trivial `ppc-params.json` alongside the profile?
```json
{
   "reserved-cpus": "0-3",
   "split-reserved-cpus-across-numa-nodes": false,
   ...
}
```

Reply:
Yup, totally. IMO this is both simpler to code (and maintain) and to extend later. We can just add annotations to ProfileCreatorArgs, `json.Unmarshal` it and call it a day.

Reply:
+1 on having args in a separate json file!

Reply:
done